backends/graphs/graphs.h

```diff
@@ -149,6 +149,78 @@ class ControlGraphs : public Inspector {
     boost::optional<cstring> instanceName{};
 };
 
+
+class ParserGraphs : public Inspector {
+ public:
+    enum class VertexType {
+        HEADER,
+        PARSER,
+        DEFAULT,
+        STATEMENTS
+    };
+    struct Vertex {
+        cstring name;
+        VertexType type;
+    };
+    class GraphAttributeSetter;
+    // The boost graph support for graphviz subgraphs is not very intuitive. In
+    // particular the write_graphviz code assumes the existence of a lot of
+    // properties. See
+    // https://stackoverflow.com/questions/29312444/how-to-write-graphviz-subgraphs-with-boostwrite-graphviz
+    // for more information.
+    using GraphvizAttributes = std::map<cstring, cstring>;
```

Comment:
it seems you pasted a lot of code from `ControlGraphs` here, and that these 2 classes would benefit from inheriting from a common base class